ddtrace/tracer.py

```diff
@@ -357,7 +364,36 @@ def start_span(self, name, child_of=None, service=None, resource=None, span_type
             context = tracer.get_call_context()
             span = tracer.start_span('web.worker', child_of=context)
         """
-        new_ctx = self._check_new_process()
+        if is_in_child_after_fork:
+            self._pid = os.getpid()
+            ctx = self.get_call_context()
+            # The spans remaining in the context can not and will not be finished
+            # in this new process. So we need to copy out the trace metadata needed
+            # to continue the trace.
+            # Also, note that because we're in a forked process, the lock that the
+            # context has might be permanently locked so we can't use ctx.clone().
+            new_ctx = Context(
+                sampling_priority=ctx._sampling_priority,
+                span_id=ctx._parent_span_id,
+                trace_id=ctx._parent_trace_id,
+            )
+            self.context_provider.activate(new_ctx)
```

Comment:
This can be done in `after_in_child` AFAIU?

Reply:
There's no `self` reference available in the `after_in_child` and there's no easy way (that I can see) to achieve that since the decorator is applied at load-time. This led to the not-great argument introspection solution.

Reply:
It's up to the object in its `__init__` function to call `after_in_child` with a bound-method then.